drivers/pds/api/backupCredential.go

```diff
@@ -0,0 +1,204 @@
+package api
+
+import (
+	"context"
+	status "net/http"
+
+	pds "github.com/portworx/pds-api-go-client/pds/v1alpha1"
+	log "github.com/sirupsen/logrus"
+)
+
+// BackupCredential struct
+type BackupCredential struct {
+	context   context.Context
+	apiClient *pds.APIClient
+}
+
+// ListBackupCredentials func
+func (backupCredential *BackupCredential) ListBackupCredentials(tenantID string) ([]pds.ModelsBackupCredentials, error) {
+	backupClient := backupCredential.apiClient.BackupCredentialsApi
+	backupModels, res, err := backupClient.ApiTenantsIdBackupCredentialsGet(backupCredential.context, tenantID).Execute()
+
+	if res.StatusCode != status.StatusOK {
+		log.Errorf("Error when calling `ApiTenantsIdBackupCredentialsGet``: %v\n", err)
+		log.Errorf("Full HTTP response: %v\n", res)
+	}
+	return backupModels.GetData(), err
+}
+
+// GetBackupCredential func
+func (backupCredential *BackupCredential) GetBackupCredential(backupCredID string) (*pds.ModelsBackupCredentials, error) {
+	backupClient := backupCredential.apiClient.BackupCredentialsApi
+	backupModel, res, err := backupClient.ApiBackupCredentialsIdGet(backupCredential.context, backupCredID).Execute()
+
+	if res.StatusCode != status.StatusOK {
+		log.Errorf("Error when calling `ApiBackupCredentialsIdGet``: %v\n", err)
+		log.Errorf("Full HTTP response: %v\n", res)
+	}
+	return backupModel, err
+}
+
+// CreateAzureBackupCredential func
+func (backupCredential *BackupCredential) CreateAzureBackupCredential(tenantID string, name string, accountKey string, accountName string) (*pds.ModelsBackupCredentials, error) {
+	backupClient := backupCredential.apiClient.BackupCredentialsApi
+	azureCredsModel := pds.ModelsAzureCredentials{
+		AccountKey:  &accountKey,
+		AccountName: &accountName,
+	}
+	controllerCreds := pds.ControllersCredentials{
+		Azure: &azureCredsModel,
+	}
+	createRequest := pds.ControllersCreateBackupCredentialsRequest{
+		Credentials: &controllerCreds,
+		Name:        &name,
+	}
+	backupModel, res, err := backupClient.ApiTenantsIdBackupCredentialsPost(backupCredential.context, tenantID).Body(createRequest).Execute()
+	if res.StatusCode != status.StatusOK {
+		log.Errorf("Error when calling `ApiTenantsIdBackupCredentialsPost``: %v\n", err)
+		log.Errorf("Full HTTP response: %v\n", res)
+	}
+	return backupModel, err
+
+}
+
+// CreateS3BackupCredential func
+func (backupCredential *BackupCredential) CreateS3BackupCredential(tenantID string, name string, accessKey string, endpoint string, secretKey string) (*pds.ModelsBackupCredentials, error) {
```

Comment:
if these params are common to all functions, can we set them as part of struct itself

Reply:
These params are not common to all functions hence, didn't added as part of the struct.